neo/d3xp/script/Script_Interpreter.cpp

```diff
@@ -176,135 +178,166 @@ void idInterpreter::Reset( void ) {
-		if ( scope == &def_namespace ) {
-			return false;
-		}
-
-		d = gameLocal.program.GetDef( NULL, name, scope );
-		if ( !d ) {
-			d = gameLocal.program.GetDef( NULL, name, &def_namespace );
-			if ( !d ) {
-				return false;
+	d = gameLocal.program.GetDef(NULL, name, scope);
+
+	// Check the objects for it if it wasnt local to the function
+	if (!d)
+	{
+		for (; scopeObj && scopeObj->TypeDef()->SuperClass(); scopeObj = scopeObj->TypeDef()->SuperClass()->def)
+		{
+			d = gameLocal.program.GetDef(NULL, name, scopeObj);
+			if (d)
+			{
+				break;
 			}
 		}
 	}
 
-	reg = GetVariable( d );
-	switch( d->Type() ) {
+	if (!d)
+	{
+		out = "???";
+		return false;
+	}
+
+	reg = GetVariable(d);
+	switch (d->Type()) {
 	case ev_float:
-		if ( reg.floatPtr ) {
-			out = va("%g", *reg.floatPtr );
-		} else {
+		if (reg.floatPtr) {
+			out = va("%g", *reg.floatPtr);
+		}
+		else {
 			out = "0";
 		}
 		return true;
 		break;
 
 	case ev_vector:
-		if ( reg.vectorPtr ) {
-			out = va( "%g,%g,%g", reg.vectorPtr->x, reg.vectorPtr->y, reg.vectorPtr->z );
-		} else {
+		if (reg.vectorPtr) {
+			out = va("%g,%g,%g", reg.vectorPtr->x, reg.vectorPtr->y, reg.vectorPtr->z);
+		}
+		else {
 			out = "0,0,0";
 		}
 		return true;
 		break;
 
 	case ev_boolean:
-		if ( reg.intPtr ) {
-			out = va( "%d", *reg.intPtr );
-		} else {
+		if (reg.intPtr) {
+			out = va("%d", *reg.intPtr);
+		}
+		else {
 			out = "0";
 		}
 		return true;
 		break;
 
 	case ev_field:
-		if ( scope == &def_namespace ) {
+	{
+		idEntity* entity;
+		idScriptObject* obj;
+
+		if (scope == &def_namespace) {
 			// should never happen, but handle it safely anyway
 			return false;
 		}
 
-		field = scope->TypeDef()->GetParmType( reg.ptrOffset )->FieldType();
-		obj   = *reinterpret_cast<const idScriptObject **>( &localstack[ callStack[ callStackDepth ].stackbase ] );
-		if ( !field || !obj ) {
+		field = d->TypeDef()->FieldType();
+		entity = GetEntity(*((int*)&localstack[localstackBase]));
+		if (!entity || !field)
+		{
+			return false;
+		}
+
+		obj = &entity->scriptObject;
+		if (!obj) {
 			return false;
 		}
 
-		switch ( field->Type() ) {
+		switch (field->Type()) {
 		case ev_boolean:
-			out = va( "%d", *( reinterpret_cast<int *>( &obj->data[ reg.ptrOffset ] ) ) );
+			out = va("%d", *(reinterpret_cast<int*>(&obj->data[reg.ptrOffset])));
 			return true;
 
 		case ev_float:
-			out = va( "%g", *( reinterpret_cast<float *>( &obj->data[ reg.ptrOffset ] ) ) );
+			out = va("%g", *(reinterpret_cast<float*>(&obj->data[reg.ptrOffset])));
 			return true;
 
+		case ev_string: {
+			const char* str;
+			str = reinterpret_cast<const char*>(&obj->data[reg.ptrOffset]);
+			if (!str) {
+				out = "\"\"";
+			}
+			else {
+				out = "\"";
+				out += str;
+				out += "\"";
+			}
+			return true;
+		}
+
 		default:
 			return false;
 		}
+
 		break;
+	}
 
 	case ev_string:
-		if ( reg.stringPtr ) {
+		if (reg.stringPtr) {
 			out = "\"";
 			out += reg.stringPtr;
 			out += "\"";
-		} else {
+		}
```

Comment:
your autoformatter settings are real weird - either you put the braces on the same line as `else` or not, but this looks asymmetric :-P

Reply:
yea again man see provious comment. something is wierd.
this behaviour can easily be controlled by an 3rd party cross IDE solution